File: Android/app/src/main/java/com/diveapp/android/model/Enums.kt
package com.diveapp.android.model

import kotlinx.serialization.Serializable

@Serializable
enum class AuthProvider {
    NAVER,
    GOOGLE,
}

@Serializable
enum class UserRole {
    USER,
    ADMIN,
}

@Serializable
enum class AccountStatus {
    ACTIVE,
    DORMANT,
    SUSPENDED,
    DELETED,
}

@Serializable
enum class DiveType {
    FREEDIVING,
    SCUBA,
}

@Serializable
enum class CertificationOrganization {
    AIDA,
    PADI,
    SSI,
    RAID,
    SDI,
    NAUI,
    CMAS,
}

@Serializable
enum class BannerType {
    NOTICE,
    EVENT,
    PROMOTION,
    INFORMATION,
}

File: Backend/app/models/enums.py
import enum


class AccountStatus(str, enum.Enum):
    ACTIVE = "ACTIVE"
    DORMANT = "DORMANT"
    SUSPENDED = "SUSPENDED"
    DELETED = "DELETED"


class AuthProvider(str, enum.Enum):
    NAVER = "NAVER"
    GOOGLE = "GOOGLE"
    EMAIL = "EMAIL"


class UserRole(str, enum.Enum):
    USER = "USER"
    ADMIN = "ADMIN"


class DiveType(str, enum.Enum):
    FREEDIVING = "FREEDIVING"
    SCUBA = "SCUBA"


class CertificationOrganization(str, enum.Enum):
    AIDA = "AIDA"
    PADI = "PADI"
    SSI = "SSI"
    RAID = "RAID"
    SDI = "SDI"
    NAUI = "NAUI"
    CMAS = "CMAS"


class BannerType(str, enum.Enum):
    NOTICE = "NOTICE"
    EVENT = "EVENT"
    PROMOTION = "PROMOTION"
    INFORMATION = "INFORMATION"
